### DISCUSSION TITLE
Passthrough a bearer token to a REST API

### DISCUSSION THREAD

**Discussion by mcalder-ccs** [Budibase/budibase#13241] [Help] [answered]
Hi folks, I'm in the process of doing a technical PoC using Budibase, and I've come up against a real stumper.  I want to use BBase to build simple front-ends for my secured back-end APIs.  These APIs all require bearer tokens to use them.  Additionally, because of zero trust, each user authenticates to the platform and receives their own token.

Using BBase, I can set up a REST-based data provider, and I can pass a token to each query, but it's the same token that BBase stores during design and passes it with every request.  I can't even use a binding to set the value of the bearer token to a stored state variable.  It doesn't look like there is any way to pass user-specific bearer tokens to REST APIs that are being used as data sources.

Please tell me that there is something somewhere in the documentation that I've missed.  Otherwise, BBase is a non-starter for us and that would make me sad.

**Comment by andz-bb** [comment, marked as answer]
hey @mcalder-ccs, this isn't possible with the native authentication functionality for the REST API datasource, but can be achieved by adding an `Authorization` header in the following format:

<img width="909" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/ca161c89-291d-421c-92b5-32fd7d4d8e58">

If you set up a REST query binding for the token, you'll be able to get the relevant token for the current user when configuring the REST query in the design section, and then pass it as a binding to be used in the request header.

<img width="949" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/3602a0e3-83bd-4785-8392-04e1f3ef7ded">

Hope this helps - if you have any questions at all on this let me know!

**Reply by mcalder-ccs** [reply]
Thanks for your reply Andrew!  I was able to use {{Binding.token}} to pass the token into the API call, but I'm getting hung up on getting the token from the form into the binding.  I blame it on my lack of BBase experience.
I have a "Welcome" form where I ask a user to enter their token.  That form stores the value in a State variable.  But when I use {{State.token}} or {{frmWelcome.Fields.txtToken}} the API query doesn't work.  I know the query itself is fine because it works with the bound token when I just paste the value in, but not when I'm referencing anything else.  Any thoughts on that?

**Reply by mcalder-ccs** [reply]
Hello again!  I found out how to pass in the token from a State value as a binding.  Andrew's answer above was spot on, I just hadn't figured out how to get to the screen that he had as his second illustration.  All is good now.  Thanks again, Andrew!
